src/Razor/src/Microsoft.VisualStudio.RazorExtension/Snippets/SnippetService.cs

```diff
@@ -149,11 +163,12 @@ private ImmutableArray<SnippetInfo> ExtractSnippetInfo(SnippetLanguage language,
             // Allocate enough memory for one VSExpansion structure. This memory is filled in by the Next method.
             pSnippetInfo[0] = Marshal.AllocCoTaskMem(Marshal.SizeOf(snippetInfo));
 
-            var langGuid = language == SnippetLanguage.CSharp
-                ? s_CSharpLanguageId
-                : s_HtmlLanguageId;
+            var inboxSnippets = language == SnippetLanguage.CSharp
+                ? s_inboxSnippets[s_CSharpLanguageId]   // C# in box snippets are handled by the C# language service
+                : snippetSetting == SnippetSetting.All
+                    ? ImmutableHashSet<string>.Empty
+                    : s_inboxSnippets[s_HtmlLanguageId];
```

Comment:
This was hard to parse, can you separate it out for us old folk.

If we're not ignoring all snippets for C# when the setting is "All", then how does "All" and "Custom" differ? I would have expected the "All" condition be first. Also `ignoredSnippets` would be a clearer name.